### PR TITLE
Improvements for the SQS portion of the library

### PR DESCRIPTION
**Pull request by scohen**
This fixes a bug in the list_queues function as well as adding parsing for all XML responses. 
Added unit tests for each response type, and I'll add more for errors as I encounter them.

@benwilson512

### CODE REVIEW

File: lib/ex_aws/sqs.ex [hunk 1]
```diff
@@ -259,10 +262,16 @@ defmodule ExAws.SQS do
   end
 
   defp request(queue, action, params) do
+    action_atom = action
+    |> Macro.underscore
+    |> String.to_atom
```
Comment (benwilson512) [inline lib/ex_aws/sqs.ex]: Would it be preferable to have for example `request("", :list_queues, params)` and then

```
defp request(queue, action_atom, params) do
  action = action_atom |> Atom.to_string |> Macro.camelize
end

File: test/lib/ex_aws/sqs/parser_test.exs [hunk 2]
```diff
@@ -0,0 +1,434 @@
+defmodule ExAws.SQS.ParserTest do
+  use ExUnit.Case, async: true
```
Comment (benwilson512) [inline test/lib/ex_aws/sqs/parser_test.exs]: You're the best person on earth for including all of these tests.

File: lib/ex_aws/sqs/parsers.ex [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+
+      {:ok, Map.put(resp, :body, parsed_body)}
+    end
+
+    def attribute_list_to_map(list_of_maps) do
+      list_of_maps
+      |> Enum.reduce(%{}, fn(%{name: name, value: val}, acc) ->
+        attribute_name = name
+        |> Macro.underscore
+        |> String.to_atom
```
Comment (benwilson512) [inline lib/ex_aws/sqs/parsers.ex]: The other `String.to_atom` call is fine cause all the inputs are basically hardcoded.

Aren't attribute lists basically arbitrary input?

File: lib/ex_aws/sqs/parsers.ex [hunk 4]
```diff
@@ -0,0 +1,204 @@
+if Code.ensure_loaded?(SweetXml) do
+  defmodule ExAws.SQS.Parsers do
+    import SweetXml, only: [sigil_x: 2]
+
+    def parse({:ok, resp=%{body: xml}}, :list_queues) do
```
Comment (benwilson512) [inline lib/ex_aws/sqs/parsers.ex]: I'm gonna be "that guy" and note that as a stylistic thing I tend to prefer `%{body: xml} = resp`

File: lib/ex_aws/sqs.ex (hunk 1 above)
Comment (scohen) [inline lib/ex_aws/sqs.ex]: Ok. I was trying to make as few changes as possible, but atoms are great.

File: test/lib/ex_aws/sqs/parser_test.exs (hunk 2 above)
Comment (scohen) [inline test/lib/ex_aws/sqs/parser_test.exs]: I like tests. A lot.

File: lib/ex_aws/sqs/parsers.ex (hunk 4 above)
Comment (scohen) [inline lib/ex_aws/sqs/parsers.ex]: I don't have a preference. NBD.

File: lib/ex_aws/sqs/parsers.ex (hunk 3 above)
Comment (scohen) [inline lib/ex_aws/sqs/parsers.ex]: Oh, derp.

**Pull request merged**